app/src/ui/branches/PublishBranch.tsx

```diff
@@ -0,0 +1,58 @@
+import * as React from 'react'
+import { Dispatcher } from '../../lib/dispatcher'
+import { Branch } from '../../models/branch'
+import { ButtonGroup } from '../../ui/lib/button-group'
+import { Button } from '../../ui/lib/button'
+import { Dialog, DialogContent, DialogFooter } from '../../ui/dialog'
+import { Repository } from '../../models/repository'
+
+interface IPublishBranchProps {
+  readonly dispatcher: Dispatcher
+  readonly repository: Repository
+  readonly branch: Branch
+  readonly onPushed: (repository: Repository, branch: Branch) => void
+  readonly onDismissed: () => void
+}
+
+export class PublishBranch extends React.Component<IPublishBranchProps> {
+  public render() {
+    return (
+      <Dialog
+        id="publish-branch"
+        key="publish-branch"
+        title={__DARWIN__ ? 'Publish Branch' : 'Publish branch'}
+        onDismissed={this.cancel}
+        onSubmit={this.cancel}
+      >
+        <DialogContent>
+          <p>
+            Your branch must be published before opening a pull request. Would
+            you like to publish <b>{this.props.branch.name}</b> and oepn a pull
+            request?
+          </p>
+        </DialogContent>
+
+        <DialogFooter>
+          <ButtonGroup>
+            <Button type="submit">Cancel</Button>
```

Comment:
I think this button order is off. The Cancel button should be on the left and the `ButtonGroup` will need the `destructive` prop.